libs/barista-components/chart/src/selection-area/selection-area-overlay-ref.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { OverlayRef } from '@angular/cdk/overlay';
+import { Observable, Subject } from 'rxjs';
+import { DtChartSelectionAreaContainer } from './selection-area-container';
+import { OnDestroy } from '@angular/core';
+
+export class DtChartSelectionAreaOverlayRef implements OnDestroy {
```

Comment:
Please remove implements as it is a normal class and not a Component or Service – therefore you have No Angular Lifecycle methods.